src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/api/jira/DataCenterFormula.kt

```diff
@@ -3,7 +3,6 @@ package com.atlassian.performance.tools.awsinfrastructure.api.jira
 import com.amazonaws.services.cloudformation.model.Parameter
 import com.atlassian.performance.tools.aws.api.*
 import com.atlassian.performance.tools.awsinfrastructure.ApplicationStorageWrapper
-import com.atlassian.performance.tools.awsinfrastructure.InstanceAddressSelector
```

Comment:
Is `InstanceAddressSelector` used anywhere anymore? I'm not sure if it was covered by tests, but do we consciously avoid that behavior now?

Reply:
I believe it's not used in this repository anymore. It's not `internal` though, despite not being in the `com.atlassian.performance.tools.awsinfrastructure.api.*` package, so I'm not sure if we can remove it. Maybe we should deprecate it.

It has 2 tests, but they are not telling me exactly why it was introduced and what use case it's meant to cover. 

The functionality basically tries to connect to a given port via public IP of the instance to see if it's accessable. If it fails 4 times (1s delay) then it checks the same for private IP of that instance. It will throw an exception if neither is responding.

I found our usage of it flawed. In most of the places we were calling this mechanism with default port (ssh/22) to get an address for communicating though different port - I believe we should at least check the port which we wanted to communicate though (e.g. mysql or nfs port). This mechanism also is always run on the test coordinator machine, i.e. it's not run by VU or load balancer machine, so it made little sense to me that it determines IP though which an external machine will comunicate. I think it's better to be explicit with it.

Reply:
@pbruski thoughts?

Reply:
My understanding is that we won't have public instances anymore - is that correct? If so, we don't need the address selector and can just use the private address all the time.
The class was created so that we're able to use private addresses. DCNG was only using private networking.

Reply:
Regarding the port selection, it was chosen because we use SSH to instrument the instances, so all should have that port available. That's why we also used the test coordinator.

Reply:
- [ ] Deprecate the `InstanceAddressSelector`